apps/api/src/integration/plugins/aws-lambda.integration.ts

```diff
@@ -138,13 +142,149 @@ export class AWSLambdaIntegration extends BaseIntegration {
         await this.delegateConfigurationDeletedEvent(data)
         break
 
+      case EventType.ENVIRONMENT_DELETED:
+        await this.handleEnvironmentDeleted(data)
+        break
+
+      case EventType.ENVIRONMENT_UPDATED:
+        await this.handleEnvironmentUpdated(data)
+        break
+
       default:
         this.logger.warn(
           `Event type ${data.eventType} not supported for AWS Lambda integration.`
         )
     }
   }
 
+  private async handleEnvironmentDeleted(
+    data: IntegrationEventData
+  ): Promise<void> {
+    this.lambda = this.getLambdaClient()
+
+    const integration = this.getIntegration<AWSLambdaIntegrationMetadata>()
+    const environmentId = data.event.itemId
+
+    // AWS Lambda integration supports only a single environment
+    const attachedEnv = integration.environments?.[0]
+    if (!attachedEnv || attachedEnv.id !== environmentId) {
+      this.logger.log(
+        `Environment ${environmentId} not attached to this Lambda integration. Idempotent success.`
+      )
+      return
+    }
+
+    const { id: integrationRunId } = await this.registerIntegrationRun({
+      eventId: data.event.id,
+      integrationId: (integration as any).id,
+      title: `Cleaning up Lambda environment ${attachedEnv.slug}`
+    })
+
+    try {
+      // Fetch variable and secret names that belong to this environment
+      const variableVersions = await (this as any).prisma.variableVersion.findMany({
+        where: { environmentId },
+        distinct: ['variableId'],
+        include: { variable: true }
+      })
+
+      const secretVersions = await (this as any).prisma.secretVersion.findMany({
+        where: { environmentId },
+        distinct: ['secretId'],
+        include: { secret: true }
+      })
+
+      const keysToRemove = new Set<string>()
+      variableVersions.forEach((v) => keysToRemove.add(v.variable.name))
+      secretVersions.forEach((s) => keysToRemove.add(s.secret.name))
+
+      // Fetch existing env vars from Lambda
+      const { existingEnvironmentalValues } = await this.getAllEnvironmentalValues(
+        integration.metadata.lambdaFunctionName
+      )
+
+      let changed = false
+      for (const key of Array.from(keysToRemove)) {
+        if (existingEnvironmentalValues.has(key)) {
+          existingEnvironmentalValues.delete(key)
+          changed = true
+        }
+      }
+
+      if (changed) {
+        await retryWithBackoff(() =>
+          this.updateLambdaFunctionConfiguration(
+            integration.metadata.lambdaFunctionName,
+            existingEnvironmentalValues
+          )
+        )
+      }
+
+      // Disconnect environment from integration
+      await (this as any).prisma.integration.update({
+        where: { id: (integration as any).id },
+        data: { environments: { disconnect: { id: environmentId } } }
+      })
+
+      await this.markIntegrationRunAsFinished(
+        integrationRunId,
+        IntegrationRunStatus.SUCCESS,
+        0,
```

Comment:
Update this line to properly update the duration. check the other functions to understand how

Reply:
@johnny603 you forgot this one